From 296f6ab3a619b47b6f8d4ac42ec704a700df08b2 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 8 Mar 2022 12:47:49 +0000
Subject: [PATCH] ARM: dts: bcm2711-rpi-ds: Disable the BCM2835 STC

Although BCM2711 still includes the old BCM2835 system timer, the newer
per-core local timers are preferred because they are more efficient to
access and can generate core-specific interrupts. Make the usage clear
by disabling the driver for the old STC.

See: https://github.com/raspberrypi/firmware/issues/1702

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711-rpi-ds.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
@@ -203,3 +203,7 @@
 &dvp {
 	status = "disabled";
 };
+
+&system_timer {
+	status = "disabled";
+};
